### DISCUSSION TITLE
解雇出store中的数据，在set数据后，不会重新渲染页面

### DISCUSSION THREAD

**Discussion by KunHuoJ** [vuejs/pinia#2520] [Help and Questions]
![image](https://github.com/vuejs/pinia/assets/118970817/b1bb4d08-8499-47bf-8101-6d5e3f9a3836)
我从store中解构出userinfo 和 一个setUserInfo方法
![image](https://github.com/vuejs/pinia/assets/118970817/b6253243-5792-4095-bce2-aa0de461cba2)
我在页面中使用他
然后用户点击修改信息后会重新setUserInfo 最新的数据进store ，但是我userInfo.username 不会改变，必须刷新一下页面才会变成最新值，但是我如果采用以下写法，不结构直接使用就可以，这是为什么呢
![image](https://github.com/vuejs/pinia/assets/118970817/1db448d3-7da5-4e8f-b744-cc45155c37ee)

**Comment by jzs1123** [comment]
解构会失去响应式，使用storeToRefs保持响应式
const { userName, role} = storeToRefs(useDatas())
